packages/repository/src/decorators/repository.ts

```diff
@@ -77,11 +77,11 @@ export class RepositoryMetadata {
  * - @repository('Customer', mysqlDataSource)
  * - @repository(Customer, 'mysqlDataSource')
  */
-export function repository(model: string | Class<Model>,
+export function repository<T extends Model>(model: string | Class<T>,
```

Comment:
@raymondfeng I feel it's better to decide what is the model type of the repository early on, so that the target descriptor type (see below) can expect the target property to have the same repository/model type. Thoughts?